docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,230 @@
+= Kubernetes
+
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+Kubernetes is an ideal environment for deploying and running highly available services.
+This section describes the specifics of deploying HA applications built on WildFly in this cloud environment.
+
+== Role of the Distributed Cache Mode
+
+WildFly uses the project <<Infinispan>> for the distributed cache implementation.
+While there are multiple cache modes available, the 'distributed' cache is the optimal one for the cloud environments.
+This cache mode stores data in a configurable number of cluster members (defaulting to `2`),
+and thus allowing users to elect the desired balance of required redundancy and performance.
+This cache mode relies on consistent hashing to determine the 'primary owner' and the 'backup owners' of given cache entries.
+In order to achieve optimal performance, the load balancer deployed in front of the application server cluster:
+
+1. must route HTTP requests for a given session to the correct primary owner,
+2. as the topology changes, the application server must be able to update the affinity information.
+
+This can be done by configuring HAProxy Ingress Controller described below.
+
+=== Changing Cache Mode in Deployment Image
+
+// TODO - update the wording when we make 'dist' the actual cloud default - https://issues.redhat.com/browse/CLOUD-4211
+Presently, existing applications need to update the default cache mode to optimize scalability.
```

Comment:
We should explain that the default configuration currently uses replicated caches, which effectively fixes the heap capability of the cluster such that it cannot be increased by scaling up.

Reply:
Added.